### DISCUSSION TITLE
The right hardware

### DISCUSSION THREAD

**Discussion by Nickduino** [Ralim/IronOS#1913] [General]
I was pretty set on getting a Pinecil V2 until I saw that review, especially the part where he tries large ground planes:
https://www.youtube.com/watch?v=-TosWhPsru4&t=766s

I first thought he was simply using the wrong power supply that couldn't deliver enough power but if you look at the banner at the top of the screen, it shows it was able to pull 32 W earlier. And at 15:04 he says the soldering iron is reading 350°C, which would not indicate a power delivery issue. And this happens to him with different tips.

Someone on Reddit says it's because of the thermal gradient in the tip and the fact the thermocouple shares its line with the heater but I fail to why the iron would still display 350°C when it is realistically much colder nor why the temp reading would be so wrong while the duty cycle of the heating element is so low (because it's pulling under 10W).

**Do you know what actually happen during his review? Would another iron running IronOs fare better? One that uses other tips that don't share their thermocouple line with the heater element (if that exists)?**

**Comment by Ralim** [comment]
Hello,

There is a fair bit going on here. Let me know if the below helps or just makes it more confusing :)
I've described the main two aspects. Let me know if you want more detail on any of it but this is about all I have time to write at the moment.

Before I go into this, all modern Hakko Tips, JBC Tips, Weller tips, TS100/TS101/TS80/TS80P/Pinecil/PinecilV2/S99/S80 etc tips are the same for all of this. Just varying max power limits / finer details. The actual thermal gradient stuff applies to all of them. A bit portion of this is using a tip that is right sized for the job.

If you take apart one of these tips, it looks loosely like this (best image I can find right now).
This is for the older Miniware style tips rather than the "short" tips that I customised for the PinecilV2.
![image](https://github.com/Ralim/IronOS/assets/5425387/242d23c9-6593-4a3f-821c-68a7a2a1952f)

Key parts that matter here:
Bottom right is the copper slug that forms the "inside" of the tip.
Bottom middle is the heater coil + thermocouple assembly.

Here is a marking cutaway as well for this style of tip, which helps with visualising hopefully :)
![image](https://github.com/Ralim/IronOS/assets/5425387/55f715ab-8902-4656-998a-358fb89e3cde)

At the bottom of the heater coil you can see a small dot/blob. This is the join that forms the thermocouple.
So this is where the iron is measuring the temperature. This is _far more_ accurate than the older style of tip where you had a metal tip that goes "over" a heater core. As now the temperature sensor is bonded to the inner copper slug that forms the heat spreader in the tip. This is not as accurate as the rf based tip heating/sensing tips ( though that is super expensive, and heavily patented).

So what IronOS is doing, is reading that thermocouple and then regulating the heater to hold it at the set temperature as best it can.


>Someone on Reddit says it's because of the thermal gradient in the tip and the fact the thermocouple shares its line with the heater 

So this is largely right in that its thermal gradient of the tip and the location of the thermocouple. How the thermocouple is wired up doesnt matter for this.

#### Wiring up thermocouple inline vs separate 

Wiring up the thermocouple as an inline joined has some pro's and cons:

Pros:
1. Far, far cheaper
2. More reliably aligned (cant be miss positioned in when cementing in the heater core
3. Easier for factory to test
4. Lets you get the temperature sensor reasonably close to the tip by making it the leading edge of the coil

Cons:
1. You need to stop heating the coil to measure the temperature
2. Uses a thermocouple (instead of say a thermistor / PT100), so needs cold junction offset handling

In the case of the PinecilV2 on the short tips we make the coil more dense towards the tip (to focus heat at the front) which improves performance and reduces heat loss to the side of the tip.

In terms of turning off the heater, because its on the same wires what endsup happening is our cycle looks like

1. Turn on the heater for up to 255 ticks (always lasts 255 ticks long, on time varies by PID)
2.  Turn off heater for 10 ticks
3. Take temperature measurement (takes 10 ticks)
4. Repeat

So this means that we can only ever have the heater on for 93% of the time.
When we are actually driving the tip hard, (such as on first heat up) we actually slow this down and take the temperature reading half as often, so this means that we have a maximum heater on time of 96%. (This is why you can see the screen temp updates slower during the initial heat and then speeds up).

This also means that if you look at the estimated output power, if the iron is driving at less than those numbers, then you are **not** limited by this at all but instead thermal constraints. When the iron is locked at those percentages, then you are limited by the way we measure the temp (with the off cycle). In practical use, the vast majority of the time you are running at 20-50W so you dont run into this headroom as much as you expect except for very large ground planes using a large tip.

I list this as a con, as it means we cant continuously measure and adjust the drive current while running. However for these small tips that can thermally saturate rapidly, I would argue you _should_ turn off the heater to measure anyway so that the temperature more accurately reflects the outside tip temperature. But this is a discussion that sort of goes in circles depending on what you care more about (time to soldarable temperature or small temperature delta tracking).

If you want to go separate temp sensor you will likely end up at the old vintage heater-cartridge and separate tip style instead. These have a nice thermal barrier of air in their assembly and generally perform worse 😁  or you end up with an RF resonator soldering iron (ersa for example) which are fantastic but often rather pricey.

### Thermal gradients

This is generally the number 1 issue I see with lots of the "this doesnt solder to X" problems.

The heater coil is cemented into the inner copper slug. This generally results in the temperature of the inner slug is what the iron is regulating and controlling. So to get your board hot enough to solder to, you need to get the heat from that copper slug into your board as efficiently as possible.


To get a good solder joint, all of the parts you are soldering to need to be bought up to temperature.
The iron is regulating that heater core as tightly as it can to your setpoint.

Generally we have a chain of thermal resistances:

```mermaid
flowchart TD
    A[Heater Core] -->|Copper Slug| B(Tip Outside)
    B -->|Contact Area 1|C(Solder)
    C -->|Contact Area 2|D(PCB)
```

The copper slug thermal resistance is small; for most cases can be assumed to be 0.

The next two stages are usually where the limiting factor comes in.

The ability of the tip to transfer its heat into your PCB depends on where on the tip you are contacting, and the surface area of that segment. Then you want a _molten_ solder puddle to thermally bridge to the PCB. Then you have to contend with the PCB acting as a thermal heatsink.

So things you should do to improve this thermal contact:

Hold still, don't move around. Hold in one spot with modest pressure against the board (to keep the solder layer thing to reduce thermal resistance). Wait for the board to come up to temperature. Waving the iron around by moving it back and forth in the puddle isn't really going to help. Your just wasting heat by spreading it out in general. And often this can lead to building up a layer of solder thats not well bonded to the PCB. This becomes more of a thermal insulator and makes it harder to use later on (13:30 ish in the video). Instead of moving like that you should lay the tip down flat (see below), with some solder and wait for the PCB to come up to temperature.
If actually doing this for real life rather than a "demo" for YouTube views, normally one would use an IR preheater or hot air to thermally warm up the board near the part to 100 ish C first. This drastically improves solderability.


Use the side of the conical tip, or the flat edge of the larger K tip. Do NOT use the "tip" of the smaller tips, as this is a long way thermally from the heater. You can see this at 13:00 in the video. The puddle melts a lot better when they are using the side of the tip, rather than trying to use the top of the tip. 

Use the right sized tip. Thermal transfer is related to the cross sectional area in contact. So if the contact area of your tip is smaller than the area you are trying to heat you are leaving performance on the table. If in doubt, use as large a tip with bigger solder bridge that will fit easily. If I'm soldering XT90 connectors to a board I'm going to use the largest tip I have. Where as for a small flat flex connector I'll go down in size until its a good fit.

Locate your tip as close to the centre of the area to heat as you can. Heat spreads outwards, so trying to heat a large pad in the middle will work out better than heating on the edge. This is doubly true on PCB's where the edges of the pad often is where the copper leads off to other traces (copper conducts heat well), so if you try and heat near a large trace your wasting heat sending it down the trace to some extent. If soldering something large like a connector, heating it up in a central location can help.


If the iron is showing a lower wattage than max (and thus holding temp) but your part isnt hot enough to solder it means that your part is absorbing all of the heat coming across through the thermal transfer. So the best thing to do is increase the thermal transfer.
This is generally something around practise as well.


### Notes

The short tips were designed to "fix" a few of the above issues by:
1. Heater more dense and closer to tip to reduce thermal loss to the outside air
2. Shorter stem to improve ability to apply pressure
3. Lower resistance so that when heat is needed, we can dump more of it into the tip
4. Trial of different internal construction methods to find the best for thermal bonding of the tip (I went through 5 different arrangements)

The Pinecil is compatible to all Hakko compatible tips (so includes TS100 ones). But not all are made equal or to the same resistance. I have definitely picked up ones off Aliexpress that perform notably worse, and on the one that I filed open, it appeared the heater coil was much further back on the tip than normal (about 3-5mm set back) which hinders the temp regulation.

TS100/TS101/Pinecil/Pinecilv2 etc all use the same "hardware" for the tip control, so if you put the same code on these, with the same tip and same supply you will get effectively the same results. Pinecilv2 mostly comes in at a lower cost (depending on source) and comes with a far more powerful processor so it feels "nicer" to use. It also has bluetooth / PD EPR if that matters to you.

**Comment by Nickduino** [comment]
So it's not exactly true to say the thermocouple shares its data line with the heater: the thermocouple **is** the apex of the heating coil and is electrically in series with it.
But the JBC C245 cartridges aren't constructed like that (since they have, I believe 3 contacts), are they?


> Let me know if you want more detail on any of it but this is about all I have time to write at the moment.

This si so much more than I was expecting 😳 Thank you for having taken the time to explain things in so much detail!


> If actually doing this for real life rather than a "demo" for YouTube views

He's testing all irons and soldering stations on the same PCBs and the same two pence coin, the point is to compare them and the other irons tend to fare better than the Pinecil (I watched many of his other videos to compare). **But in your opinion there was nothing wrong with his hardware or the OS control loop, it's just that he was not getting good contact between his tip and the PCB (or his tip/PCB were more oxidized than they were with the other irons he tested).**

**And the absolute best hardware for IronOS is the Pinecil V2 with the short tips.** The only advantages others might have is the bigger OLED, metal case or tip selection but nothing is better in the power-delivery department.


> PD EPR if that matters to you

It probably does not but I want it 😆
Has it been tested with Framework's EPR 180W charger? The Wiki mentions only 3 adapters (Apple and 2 Chinese ones). Framework's spec says it's USB PD 3.1 EPR standard through a Weltrend WT6676F controller.

Last question: the Pine64 silicone cables aren't PD 3.1 240 W, are they? Any reasonnably-priced silicone cable you can recommend?

**Reply by Ralim** [reply]
If I recall correctly (haven't dug sorry due to time) the JBC C245 cartridges are 2 wire + ground (similar to how Hakko/TS100 ones are also arguably 2 wire + ground if you count the grounding screw). The Weller small tips are true 3 wire. Though also build effectively in the same wire (third wire runs to the tip and joins the front of the heating coil).


I didn't go and watch the other videos, I really just watched this one on fast-forward. Does he say what firmware version he is running? Generally newer firmware can perform significantly better than some of the older ones. While I try and keep the factory test tooling up to date with releases it doesn't always happen when batches are being made. 

I didn't mention it before but I noticed he mentioned doing the CJC calibration. If you do that _wrong_ you can end up offsetting your tip and making it run cold. If your going to deviate from the default its best to cross check with a tip temperature measurement device to make sure it worked out correctly. (Unfortunately the firmware can only do so much to check as it cant tell one source of error apart from another).


>And the absolute best hardware for IronOS is the Pinecil V2 with the short tips. 

I would say so. Its the only hardware I had any say in the design of, so mistakes in it are on me (thus I'm motivated to fix them) rather than the reverse where I'm stuck to deal with whatever the manufacturer spits out. It is of course designed to hit a target price so decisions on features were made, but I like to think the end result works well.

The short tips especially I saw are definitely worth it. So even If I went for another handle I'd go for the short tips from Pine.

If there are tips that are notably missing from the pine lineup; feel free to raise discussions here and I can feed it back. (Obviously cant promise as a new tip line is expensive to start, but at the least it will be noted for any future evaluation).

>It probably does not but I want it 😆

I can respect that. The ~2 second heat up is very nice, though not much difference in use.

>Has it been tested with Framework's EPR 180W charger?

Not sure, I certainly haven't (getting a framework would be nice down here but also $$$).
I would suggest to ask in the Pinecil Community Chat, as that is where your most likely to hit the most people than anywhere else for EPR related testing. I will note that so long as its reasonably close to compliant it should either _just work_ or make an issue and I can try and help debug it with you 🙃 . 

Personally here I have 2 chinese charger "dev kits" and an Apple EPR charger that I use for regression testing.

https://ralim.github.io/IronOS/ is the docs for IronOS and related stuff, the Pine64 Wiki is not something I maintain (fyi).


>Last question: the Pine64 silicone cables aren't PD 3.1 240 W, are they? Any reasonnably-priced silicone cable you can recommend?

They are not, they are 20V/3A cables only. Though for the price I reccomend getting some anyway as they are quote good for the money and nice to use.

**Reply by Nickduino** [reply]
> I didn't mention it before but I noticed he mentioned doing the CJC calibration. If you do that _wrong_ you can end up offsetting your tip and making it run cold. If your going to deviate from the default its best to cross check with a tip temperature measurement device to make sure it worked out correctly.

He's using a thermocouple to check iron temperatures before doing his tests

![SDG Electronics - SDG #340 FNIRSI HS-02A 100W USB Soldering Iron and FNB48P USB Power Monitor  z1p6_cPkI1w - 1280x720 - 14m31s](https://github.com/Ralim/IronOS/assets/9784237/82f86c6b-c5f2-4476-8364-81425c21171c)

But you're right to mention the calibration in link with this video; there is even an issue about it: https://github.com/Ralim/IronOS/issues/1448

**Reply by Ralim** [reply]
Yeah #1448 is very true, the CJC is quite confusing. Though it has been improved since then a fair bit.

Also good to know that they are being cross checked 🙇🏼

**Reply by Nickduino** [reply]
> If there are tips that are notably missing from the pine lineup; feel free to raise discussions here and I can feed it back.

What is missing and easy to rectify is a picture or drawing of the tips available on the product pages. There is this one for the long gross tips:
![image](https://github.com/Ralim/IronOS/assets/9784237/12c4f153-50cb-4f6c-ab90-ba2bf77ee19a)

...which is perfect. But the three other tip sets don't show it. They show the back of the box with the drawings but in a resolution that doesn't let you see them.
(For reference: https://pine64.com/product/pinecil-soldering-tip-set-fine/)



As for a missing tip design, maybe one or two of these wide ones:
![image](https://github.com/Ralim/IronOS/assets/9784237/36e3a04d-2546-4714-8959-a627ed1a2127)

They're very handy to desolder some SOIC packages (and, more importantly, connectors or screens. I say more importantly because you can't use hot air to desolder a connector or screen if you want to reuse it) and are also used to clean up a BGA before reballing.

Of course, there is a myriad of tips with weird shapes that make one specific job easier but you'd sell a couple of each here and there, so they make no sense. So I had a novel idea: make a tip with a fork and a set screw that could receive and hold a 2.5 mm² solid core wire. The user would bend the wire in the shape they want to desolder a given package to make their own tip.
The solid core wire is a method I already use to heat up all pins at once: I just push a chisel tip on it. Having a tip made for that would be even more practical.

**Reply by Ralim** [reply]
>But the three other tip sets don't show it. They show the back of the box with the drawings but in a resolution that doesn't let you see them.

Have requested better photos of all the tips. 


Interesting idea on the tips, currently the only tip that is wip is a threaded setup to allow installing heatset inserts easier.
Will pass on the concept but unsure if volume will be there 😓

**Reply by Nickduino** [reply]
> the only tip that is wip is a threaded setup to allow installing heatset inserts easier.

CNC kitchen's set (and inserts) seems pretty good already. Is that your benchmark?
https://cnckitchen.store/products/ts100-101-adapter-only-lead-and-cadmium-free



Another tip that doesn't seem to be available and that is used very frequently in microsoldering is a bent conical
![image](https://github.com/Ralim/IronOS/assets/9784237/a603fc04-a497-4201-b2c8-5222b2728f11)

They're pretty versatile because they allow you at the same time great accuracy (with their thin point), fairly good heat transfer (used sideways) and small SMD desoldering (sideways as well: you can heat up both pads)

**Reply by Nickduino** [reply]
> > the only tip that is wip is a threaded setup to allow installing heatset inserts easier.
> 
> CNC kitchen's set (and inserts) seems pretty good already. Is that your benchmark? https://cnckitchen.store/products/ts100-101-adapter-only-lead-and-cadmium-free

They're out already!
Had I known, I would have delayed my order a bit to include them. Oh well, I should be able to play with the Pinecil tomorrow

**Reply by Nickduino** [reply]
> Oh well, I should be able to play with the Pinecil tomorrow

Got it. Manufactured in May 2024, I was surprised to get a firmware from 2022. It's an opportunity to familiarize myself with blisp, so I don't mind, but I wasn't expecting this.

**Comment by Nickduino** [comment]
>Has it been tested with Frameworks EPR 180W charger? The Wiki mentions only 3 adapters (Apple and 2 Chinese ones). Framework's spec says it's USB PD 3.1 EPR standard through a Weltrend WT6676F controller.
>
>Last question: the Pine64 silicone cables aren't PD 3.1 240 W, are they? Any reasonnably-priced silicone cable you can recommend?

Or do you know if Pine64 will update the PinePower and the cables to do EPR/PD 3.1?

**Reply by Ralim** [reply]
It's certainly been discussed but nothing concrete as of present.

**Reply by Nickduino** [reply]
...what about a Pinecil V3 with a larger color screen? :angel:

**Reply by Ralim** [reply]
Hahah, no.
Basically new revision wouldn't add much, and changing screen adds a _lot_ more cost because of new tooling.
Once there is a bigger feature to cause a new version (or another parts shortage) _then_ we would see if going to a colour screen is worth it

**Reply by Nickduino** [reply]
> Hahah, no. Basically new revision wouldn't add much, and changing screen adds a _lot_ more cost because of new tooling. Once there is a bigger feature to cause a new version (or another parts shortage) _then_ we would see if going to a colour screen is worth it

Just picking your brains but...what if someone released the files for a 3D printed shell that would accommodate another screen and the documentation to install it? Would you support that mod or that seems like too much hassle for very few people?
https://www.crystalfontz.com/product/cfal12836a0088-128x36-small-oled-display

**Reply by Ralim** [reply]
So to use that display would require wiring it up to the SPI pins instead of I2C (which makes it a bit more painful of an install).

My general stance on this sort of mod (i.e. one that has a very small target use case)  is that I'm happy to maintain the code as best as I can, but generally to get it included someone would need to open a PR to add the feature.
Obviously as I wouldn't have the hardware I couldn't regression test it, so it would by a YMMV situation and hopefully the person who sends it in can help maintain / improve it.
